### PR TITLE
Feature login firebase

### PR DESCRIPTION
**Pull request by at-trungnguyen**

### CODE REVIEW

**Comment by at-trungnguyen** [comment]
@at-tuannguyen Please review my PR!

File: app/src/main/java/com/app/android/ui/login/LoginActivityUI.kt [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    override fun createView(ui: AnkoContext<LoginActivity>) = with(ui) {
+        relativeLayout {
+            lparams(matchParent, matchParent)
+            backgroundColor = android.R.color.white
+
+            tvNext = textView {
+                id = R.id.login_activity_tv_next
+                textColor = ContextCompat.getColor(ctx, R.color.red_blur)
+                textSize = px2dip(dimen(R.dimen.login_tv_next_text_size))
+                text = resources.getText(R.string.login_activity_tv_next)
```
Comment (at-vinhhuynh) [inline app/src/main/java/com/app/android/ui/login/LoginActivityUI.kt]: remove this line and use `textview(R.string.login_activity_tv_next)`
And also check all this file

File: app/src/main/java/com/app/android/ui/login/LoginViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    internal fun createUser(key: String, email: String, password: String): Single<RegisterResponse> {
+        progressBarStatus.onNext(true)
+        return loginRepository.createUser(key, RegisterRequestBody(email, password))
+                .doOnSuccess {
+                    progressBarStatus.onNext(false)
+                    saveAccessToken(it.idToken)
+                }
+                .doOnError {
+                    progressBarStatus.onNext(false)
```
Comment (at-vinhhuynh) [inline app/src/main/java/com/app/android/ui/login/LoginViewModel.kt]: please move this line to `.doOnfinally`

File: app/src/main/java/com/app/android/ui/login/LoginViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+ */
+class LoginViewModel(private val loginRepository: LoginRepository, private val localRepository: LocalRepository) {
+
+    internal val progressBarStatus: BehaviorSubject<Boolean> = BehaviorSubject.create()
+
+    internal fun createUser(key: String, email: String, password: String): Single<RegisterResponse> {
+        progressBarStatus.onNext(true)
+        return loginRepository.createUser(key, RegisterRequestBody(email, password))
+                .doOnSuccess {
+                    progressBarStatus.onNext(false)
```
Comment (at-vinhhuynh) [inline app/src/main/java/com/app/android/ui/login/LoginViewModel.kt]: remove this line

File: app/src/main/java/com/app/android/ui/splash/SplashActivity.kt [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import org.jetbrains.anko.startActivity
+
+/**
+ * Copyright © 2018 AsianTech inc.
+ * Created by trung.nguyen on 2/11/18.
+ */
+class SplashActivity : BaseActivity() {
+
+    companion object {
+        val DELAY = 3000L
```
Comment (at-vinhhuynh) [inline app/src/main/java/com/app/android/ui/splash/SplashActivity.kt]: Please add `const`

File: app/src/main/java/com/app/android/ui/splash/SplashActivityUI.kt [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+/**
+ * Copyright © 2018 AsianTech inc.
+ * Created by trung.nguyen on 2/11/18.
+ */
+class SplashActivityUI : AnkoComponent<SplashActivity> {
+    override fun createView(ui: AnkoContext<SplashActivity>) = with(ui) {
+        relativeLayout {
+            backgroundColor = ContextCompat.getColor(ctx, android.R.color.white)
+            textView {
+                text = resources.getString(R.string.app_name)
```
Comment (at-vinhhuynh) [inline app/src/main/java/com/app/android/ui/splash/SplashActivityUI.kt]: remove this line

**Comment by at-trungnguyen** [comment]
@at-vinhhuynh Thanks, Please check again!

**Pull request closed without merging**